os/mm/mm_heap/mm_heapinfo_parse_heap.c

```diff
@@ -105,6 +114,10 @@ void heapinfo_parse_heap(FAR struct mm_heap_s *heap, int mode, pid_t pid)
 	FAR struct mm_freenode_s *fnode;
 #endif
 
+	if (mm_check_heap_corruption(heap) != OK) {
+		ASSERT(0);
```

Comment:
ASSERT already includes if conditional so that below is better.
```c
ASSERT(mm_check_heap_corruption(heap) == OK);
```

Reply:
Done.